Sources/MongoSwift/ChangeStream.swift

```diff
@@ -0,0 +1,174 @@
+import mongoc
+
+/// A `ChangeStreamToken` used for wrapping the resume token.
+public struct ChangeStreamToken: Codable {
+    private let resumeToken: Document
+
+    public init(resumeToken: Document) {
+        self.resumeToken = resumeToken
+    }
+
+    public init(from decoder: Decoder) throws {
+      self.resumeToken = try Document(from: decoder)
+    }
+}
+
+/// A MongoDB ChangeStream.
+/// - SeeAlso: https://docs.mongodb.com/manual/changeStreams/
+public class ChangeStream<T: Codable>: Sequence, IteratorProtocol {
+  /// The cached resume token.
```

Comment:
Likewise some of the documentation added to the definition of `ChangeStreamToken` can be reworded and used here to better describe what this is / what it's used for.